sql/src/main/java/io/crate/planner/operators/WindowAgg.java

```diff
@@ -102,6 +104,43 @@ private WindowAgg(LogicalPlan source, WindowDefinition windowDefinition, List<Wi
         this.standalone = standalone;
     }
 
+    @Override
+    public LogicalPlan pruneOutputsExcept(Collection<Symbol> outputsToKeep) {
+        HashSet<Symbol> toKeep = new HashSet<>(outputsToKeep);
+        toKeep.removeAll(windowFunctions);
+        List<Symbol> newStandalone = List.copyOf(toKeep);
+        for (WindowFunction windowFunction : windowFunctions) {
+            toKeep.addAll(windowFunction.arguments());
```

Comment:
Would it be possible to define a class that extends `DefaultTraversalSymbolVisitor` in oder to visit all symbols from window function, such as we have exactly the same login in `visitWindowFunction`?

Reply:
I think I now see a common pattern between the different operators.  Will try to extract some common logic to contain most of the complexity

Reply:
@kovrus pushed a fixup. Please let me know what you think.

Reply:
looks good 👍